python/ray/serve/_private/replica.py

```diff
@@ -814,145 +811,199 @@ async def call_reconfigure(self, user_config: Any):
+                        r = (request_metadata.grpc_context, r.SerializeToString())
+                    await result_queue(r)
+            elif request_metadata.is_http_request and not is_asgi_app:
+                # For the FastAPI codepath, the response has already been sent over
+                # ASGI, but for the vanilla deployment codepath we need to send it.
+                await self._send_user_result_over_asgi(result, asgi_args)
+            elif not request_metadata.is_http_request:
                 raise TypeError(
-                    f"Method '{runner_method.__name__}' returned a generator. You "
-                    "must use `handle.options(stream=True)` to call "
-                    "generators on a deployment."
+                    f"Called method '{user_method_name}' with "
+                    "`handle.options(stream=True)` but it did not return a "
+                    "generator."
                 )
-
-        except Exception as e:
-            function_name = "unknown"
-            if runner_method is not None:
-                function_name = runner_method.__name__
-            e = wrap_to_ray_error(function_name, e)
-            if request_metadata.is_http_request:
-                result = starlette.responses.Response(
-                    f"Unexpected error, traceback: {e}.", status_code=500
+        else:
+            if result_is_gen or result_is_async_gen:
+                raise TypeError(
+                    f"Method '{user_method_name}' returned a generator. "
+                    "You must use `handle.options(stream=True)` to call "
+                    "generators on a deployment."
                 )
-                await self._send_user_result_over_asgi(result, scope, receive, send)
-
-            raise e from None
-
-        if request_metadata.is_http_request and not isinstance(
-            self._callable, ASGIAppReplicaWrapper
-        ):
-            # For the FastAPI codepath, the response has already been sent over the
-            # ASGI interface, but for the vanilla deployment codepath we need to
-            # send it.
-            await self._send_user_result_over_asgi(result, scope, receive, send)
-        elif request_metadata.is_grpc_request:
-            result = (request_metadata.grpc_context, result.SerializeToString())
+            if request_metadata.is_grpc_request:
+                result = (request_metadata.grpc_context, result.SerializeToString())
 
         return result
 
-    async def call_user_method_generator(
+    async def call_user_method(
         self,
         request_metadata: RequestMetadata,
         request_args: Tuple[Any],
         request_kwargs: Dict[str, Any],
-    ) -> AsyncGenerator[Any, None]:
-        """Call a user method that is expected to be a generator.
+        *,
+        result_queue: Optional[MessageQueue] = None,
+    ) -> Any:
+        """Call a user method that is *not* expected to be a generator.
```

Comment:
Is this supposed to be for calling a user method that can be either a generator or not a generator? (got confused reading through the code, correct me if I'm wrong)

Reply:
Yes sorry this docstring hadn't been updated :)